session.go

```diff
@@ -720,6 +731,13 @@ func (q *Query) Trace(trace Tracer) *Query {
 	return q
 }
 
+// QueryObserver enables query-level observer on this query.
+// The provided observer will be called every time this query is executed.
+func (q *Query) Observer(observer QueryObserver) *Query {
```

Comment:
Should also add this to Batch

Reply:
Part of a slightly larger question - how to observe Batches? I'm not familiar with those, but from what I can see the `QueryObserver` may be called for each statement in the batch (with shared times and error) although I don't know if the rows have any meaning in batch queries.

Reply:
Batch will have multiple statements but no return values unless BatchCAS is used.